test/sink.test.js

```diff
@@ -85,6 +91,30 @@ test('.writer() - no value for "type" argument - should throw', () => {
     }).toThrowError('"type" is missing');
 });
 
+test('directory() - should fetch file in root directory', async () => {
+    const sink = new Sink({ path: './test/mock' });
+    const directoryContent = await sink.directory('/');
+    expect(directoryContent.length).toBe(1);
+    expect(directoryContent[0].fileName).toMatchSnapshot();
+    expect(directoryContent[0].content.length).toMatchSnapshot();
+});
+
+test('directory() - should fetch all files in sub directory', async () => {
+    const sink = new Sink({ path: './test/mock/' });
+    const directoryContent = await sink.directory('/sub');
+    expect(directoryContent).toMatchSnapshot();
+});
+
+test('directory() - should error when directory is missing', async () => {
+    expect.assertions(1);
+    const sink = new Sink({ path: './test/mock/' });
+    try {
+        await sink.directory('/missing');
+    } catch (e) {
+        expect(e.code).toMatchSnapshot();
```

Comment:
is this the same error behaviour/message as for other sinks?

Reply:
No - not atm - but it might have to be?

Reply:
I think so. Be nice if the error mechanism and messaging were the same across sinks

Reply:
Will fix. I guess we just want an error for "missing folder or empty result"?

Reply:
yeah that sounds good